service/matching/matcher_data.go

```diff
@@ -540,6 +578,28 @@ func (rt *resettableTimer) unset() {
 	}
 }
 
+// simple limiter
+
+// simpleLimiter implements a "GCRA" limiter. The owner should read the `ready` field directly
+// and compare to the current time to decide if a task is allowed.
+type simpleLimiter struct {
+	// parameters:
+	interval int64 // nanos/task
+	burst    int64 // burst duration in nanos
+
+	// state:
+	ready int64 // unix nanos
+}
+
+func (s *simpleLimiter) set(rate float64, burst int) {
+	s.interval = int64(float64(time.Second) / rate)
+	s.burst = s.interval * int64(burst)
+}
+
+func (s *simpleLimiter) consume(now int64, tokens int64) {
+	s.ready = max(now, s.ready+s.burst) - s.burst + tokens*s.interval
+}
```

Comment:
As discussed, additional comment would be helpful to explain the overall intuition.